crates/h265/src/config.rs

```diff
@@ -10,42 +10,94 @@ use scuffle_bytes_util::{BitReader, BitWriter};
 /// HEVC Decoder Configuration Record
 /// ISO/IEC 14496-15:2022(E) - 8.3.2.1
 pub struct HEVCDecoderConfigurationRecord {
+    /// The `configuration_version` as a u8. Matches the field as defined in ISO/IEC 23008-2.
     pub configuration_version: u8,
+    /// The `general_profile_space` as a u8. Matches the field as defined in ISO/IEC 23008-2.
     pub general_profile_space: u8,
+    /// The `general_tier_flag` as a bool. Matches the field as defined in ISO/IEC 23008-2.
     pub general_tier_flag: bool,
+    /// The `general_profile_idc` as a u8. Matches the field as defined in ISO/IEC 23008-2.
     pub general_profile_idc: u8,
+    /// The `general_profile_compatibility_flags` as a u32. Matches the field as defined in ISO/IEC 23008-2.
     pub general_profile_compatibility_flags: u32,
+    /// The `general_constraint_indicator_flags` as a u64. Matches the field as defined in ISO/IEC 23008-2.
     pub general_constraint_indicator_flags: u64,
+    /// The `general_level_idc` as a u32. Matches the field as defined in ISO/IEC 23008-2.
     pub general_level_idc: u8,
+    /// The `min_spatial_segmentation_idc` as a u16. Matches the field as defined in ISO/IEC 23008-2.
     pub min_spatial_segmentation_idc: u16,
-    pub parallelism_type: u8,
+    /// The `chroma_format_idc` as a u8. Matches the field as defined in ISO/IEC 23008-2.
     pub chroma_format_idc: u8,
+    /// The `bit_depth_luma_minus8` as a u8. Matches the field as defined in ISO/IEC 23008-2.
     pub bit_depth_luma_minus8: u8,
+    /// The `bit_depth_chroma_minus8` as a u8. Matches the field as defined in ISO/IEC 23008-2.
     pub bit_depth_chroma_minus8: u8,
+    /// The `parallelism_type` as a u8.
+    ///
+    /// 0 means the stream supports mixed types of parallel decoding or otherwise.
+    ///
+    /// 1 means the stream supports slice based parallel decoding.
+    ///
+    /// 2 means the stream supports tile based parallel decoding.
+    ///
+    /// 3 means the stream supports entropy coding sync based parallel decoding.
+    pub parallelism_type: u8,
```

Comment:
Perhaps we should create a nutype-enum for this field which has the fields u are talking about but also supports unknowns.